### PR TITLE
tdshim_spec: clarify MRTD calculation

### LINKED ISSUE #316: clarify order of MRTD calculation
The expected flow is:
 * Search TD Metadata from metadata location and get TDVF_DESCRIPTOR.
 * In TDVF_DESCRIPTOR, get TDVF_SECTION in the order of image file offset, from the lowest GPA to the highest GPA. For each TDVF_SECTION: 
   * In TDVF_SECTION, if PAGE.AUG attribute is clear (PAGE.ADD is used), extend
   MRTD with the THD.MEM.PAGE.ADD as described above, from the lowest GPA to the
   highest GPA, with 4K granularity.
   * In TDVF_SECTION, if MR.EXTEND attribute is set, extend MRTD with the
   TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
   with 256 bytes granularity.

### PR DESCRIPTION
**Pull request by longlongyang**
fix #316
Signed-off-by: Longlong Yang <longlong.yang@intel.com>

### CODE REVIEW

File: doc/tdshim_spec.md [hunk 1]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
...
    1).
-3. In TDVF_SECTION, if PAGE.AUG attribute is clear (PAGE.ADD is used), extend
-   MRTD with the THD.MEM.PAGE.ADD as described above, from the lowest GPA to the
-   highest GPA, with 4K granularity.
-4. In TDVF_SECTION, if MR.EXTEND attribute is set, extend MRTD with the
-   TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
-   with 256 bytes granularity.
+4. For every TDVF_SECTION, Walk through the TDVF_SECTION in 1 page (4096 bytes)
+   granularity in the order of index.
+5. For every page,
```
Comment (jyao1) [inline doc/tdshim_spec.md:761]: Please move 5 be a sub-item of 4.

File: doc/tdshim_spec.md [hunk 2]
```diff
@@ -170,12 +170,12 @@ The TD Shim metadata can be located at (TD Shim end – 0x20) byte. It is a
 | Value          | Name         | Memory Type       | VMM Action              | Td-Shim Action | Measurement       |
 |:---------------|:-------------|:------------------|:------------------------|:---------------|:------------------|
 | 0              | BFV          | Private Memory    | PAGE.ADD + MR.EXTEND    | N/A            | MRTD              |
-| 1              | CFV          | Private Memory    | PAGE.ADD                | RTMR.EXTEND    | RTMR[0]           |
-| 2              | TD_HOB       | Private Memory    | PAGE.ADD                | RTMR.EXTEND    | RTMR[0]           |
-| 3              | TempMem      | Private Memory    | PAGE.ADD                | N/A            | N/A               |
+| 1              | CFV          | Private Memory    | PAGE.ADD                | RTMR.EXTEND    | MRTD (and) RTMR[0]|
```
[the remark this answers is not in the document]
Comment (longlongyang) [reply doc/tdshim_spec.md:173]: > Why you change this???

The first reason is that whatever the section have PAGE.AUG cleared should be measured to MRTD, right?
The second reason is that when @liuw1 read my code, he pointed it out that there are just BFV and payload sections need to be measured to MRTD. I think there is some confusion here.

[the remark this answers is not in the document]
Comment (longlongyang) [reply doc/tdshim_spec.md]: > Why you change this???

Apology for misunderstanding, removed.

File: doc/tdshim_spec.md [hunk 3]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
...
+2. Search TD Metadata from metadata location and get TDVF_DESCRIPTOR.
+3. In TDVF_DESCRIPTOR, get TDVF_SECTION from index 0 to (NumberOfSectionEntry -
    1).
-3. In TDVF_SECTION, if PAGE.AUG attribute is clear (PAGE.ADD is used), extend
-   MRTD with the THD.MEM.PAGE.ADD as described above, from the lowest GPA to the
-   highest GPA, with 4K granularity.
-4. In TDVF_SECTION, if MR.EXTEND attribute is set, extend MRTD with the
-   TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
-   with 256 bytes granularity.
+4. For every TDVF_SECTION in the order of index, Walk through the TDVF_SECTION
```
Comment (jyao1) [inline doc/tdshim_spec.md:759]: Walk -> walk.

File: doc/tdshim_spec.md [hunk 4]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
...
+   in 1 page (4096 bytes) granularity.
+    1. For every page,
+       1. if PAGE.AUG attribute is clear (PAGE.ADD is used), extend MRTD with
```
Comment (jyao1) [inline doc/tdshim_spec.md:762]: if -> If

File: doc/tdshim_spec.md [hunk 5]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
...
+       the THD.MEM.PAGE.ADD as described above, with MemoryAddress of the
+       corresponding TDVF_SECTION + current page index * 4096 as GPA.
+       2. if MR.EXTEND attribute is set, Walk through the page in 1 chunk (256
```
Comment (jyao1) [inline doc/tdshim_spec.md:765]: if -> If
Walk -> walk

File: doc/tdshim_spec.md [hunk 6]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
...
    1).
-3. In TDVF_SECTION, if PAGE.AUG attribute is clear (PAGE.ADD is used), extend
-   MRTD with the THD.MEM.PAGE.ADD as described above, from the lowest GPA to the
-   highest GPA, with 4K granularity.
-4. In TDVF_SECTION, if MR.EXTEND attribute is set, extend MRTD with the
-   TDH.MR.EXTEND as described above, from the lowest GPA to the highest GPA,
-   with 256 bytes granularity.
+4. For every TDVF_SECTION in the order of index, Walk through the TDVF_SECTION
+   in 1 page (4096 bytes) granularity.
+    1. For every page,
```
Comment (jyao1) [inline doc/tdshim_spec.md:761]: I suggestion:
 * For every TDVF_SECTION in TDVF_DESCRIPTOR, from index 0 to (NumberOfSectionEntry -
   1)
   * For every page (4K granularity) in TDVF_SECTION, from lowest GPA to  highest GPA
     * If PAGE.AUT ...
     * If MR.EXTEND ...

File: doc/tdshim_spec.md [hunk 7]
```diff
@@ -752,15 +752,21 @@ MRTD is extended by two operations:
 the chunk’s content.
 
 The expected flow is:
-1. Search TD Metadata from metadata location and get TDVF_DESCRIPTOR.
-2. In TDVF_DESCRIPTOR, get TDVF_SECTION from index 0 to (NumberOfSectionEntry -
+1. Initialize a SHA384 hash context. Load TDVF image and parse.
+2. Search TD Metadata from metadata location and get TDVF_DESCRIPTOR.
+3. In TDVF_DESCRIPTOR, get TDVF_SECTION from index 0 to (NumberOfSectionEntry -
```
Comment (jyao1) [inline doc/tdshim_spec.md:757]: 3 and 4 are duplicated.

**Comment by longlongyang** [comment]
Updated according to the suggestion.

**Review by jyao1 — APPROVED** [review]

**Review by jyao1 — APPROVED** [review]

**Pull request merged**